### PR TITLE
feat(apigateway): Added proxy support for `organization_id_or_slug` path params

### PR DESCRIPTION
**Pull request by iamrajjoshi**
We are adding support for `{resource}.id_or_slug` support in our API path parameters. Currently our api gateway relied on "organization_slug" existing in the path parameter. 

This PR adds support for `organization_id_or_slug` existing in path parameter, so when we go change all our apis from organization_slug to organization_id_or_slug, we can still proxy correctly.

I also a test for this. Let me know if I am missing something.

Note: `id_or_slug_path_params_enabled` is a wrapper for a couple of options that are used to rollout the API changes.

### CODE REVIEW

File: src/sentry/api/utils.py [hunk 1]
```diff
@@ -498,10 +498,12 @@ def id_or_slug_path_params_enabled(
         return False
 
     # EA option for endpoints where organization is not available
-    if convert_args_class in options.get("api.id-or-slug-enabled-ea-endpoints"):
-        return True
+    if convert_args_class:
+        if convert_args_class in options.get("api.id-or-slug-enabled-ea-endpoints"):
+            return True
+        return False
```
Comment (markstory) [inline src/sentry/api/utils.py:504]:
```suggestion
        return convert_args_class in options.get("api.id-or-slug-enabled-ea-endpoints"):
```

Could golf this down.

File: src/sentry/api/utils.py [hunk 2]
```diff
@@ -498,10 +498,10 @@ def id_or_slug_path_params_enabled(
         return False
 
     # EA option for endpoints where organization is not available
-    if convert_args_class in options.get("api.id-or-slug-enabled-ea-endpoints"):
-        return True
+    if convert_args_class:
+        return convert_args_class in options.get("api.id-or-slug-enabled-ea-endpoints")
 
-    return False
+    return True
```
Comment (markstory) [inline src/sentry/api/utils.py:504]: Is this correct? If this function is called with no parameters should it return true?

Comment (iamrajjoshi) [reply src/sentry/api/utils.py:504]: If `convert_args_class` is not passed in,  but the org is passed in and in the org option, which is the case ` get_region_for_organization`, I want to return True. All the other uses of this wrapper pass in the convert_args _class.

Comment (iamrajjoshi) [reply src/sentry/api/utils.py:504]: I am going to pull this logic and add an other option.

**Review by markstory — APPROVED** [review]

**Pull request merged**